astropy/visualization/wcsaxes/coordinate_helpers.py

```diff
@@ -739,10 +743,27 @@ def set_axislabel(self, text, minpad=1, **kwargs):
         if minpad is None:
             minpad = 1
 
+        protected_kw = [
+            "x",
+            "y",
+            "rotation",
+            "horizontalalignment",
+            "ha",
+            "rotation_mode",
+        ]
+        if {*kwargs} & {*protected_kw}:
```

Comment:
while I do find your original implementation very elegant, using plain english would probably help more casual readers understanding what's happening here 
```suggestion
        if set(kwargs).intersection(protected_kw):
```